src/reanimated2/threads.ts

```diff
@@ -111,14 +128,18 @@ export function runOnUI<A extends any[], R>(
       });
     }
   };
-}
+}) as unknown as runOnUIAPI;
 
+// Check comment on runOnUIAPI type to understand this type.
+type runOnUIImmediatelyAPI = <A extends unknown[], R>(
```

Comment:
Let's use `FooType` instead of `FooAPI`.
```suggestion
type runOnUIImmediatelyType = <A extends unknown[], R>(
```

Reply:
Since this 'trick' might be used in the code multiple times I wanted to get it all under one type suffix and API is in my opinion a good one since this type actually defines how the function is perceived on the outside and how to use it. If I named it `...type` it would look just like any other TS type and might've been considered unnecessary.